bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java

```diff
@@ -426,38 +467,160 @@ private void runPanelDiscovery() {
         updateState(CHANNEL_EFFECT, new StringType(controllerInfo.getEffects().getSelect()));
+
+        @Nullable Hue stateHue = state.getHue();
+        int hue =        (stateHue !=null) ? stateHue.getValue():0;
+        @Nullable Sat stateSaturation = state.getSaturation();
+        int saturation = (stateSaturation !=null) ? stateSaturation.getValue():0;
+        @Nullable Brightness stateBrightness = state.getBrightness();
+        int brightness = (stateBrightness !=null) ? stateBrightness.getValue():0;
+
         updateState(CHANNEL_COLOR,
-                new HSBType(new DecimalType(controllerInfo.getState().getHue().getValue()),
-                        new PercentType(controllerInfo.getState().getSaturation().getValue()),
-                        new PercentType(isOn ? controllerInfo.getState().getBrightness().getValue() : 0)));
-        updateState(CHANNEL_COLOR_MODE, new StringType(controllerInfo.getState().getColorMode()));
-        if (controllerInfo.getRhythm() != null) {
-            updateState(CHANNEL_RHYTHM_ACTIVE,
-                    controllerInfo.getRhythm().getRhythmActive() ? OnOffType.ON : OnOffType.OFF);
-            updateState(CHANNEL_RHYTHM_MODE, new DecimalType(controllerInfo.getRhythm().getRhythmMode()));
-            updateState(CHANNEL_RHYTHM_STATE,
-                    controllerInfo.getRhythm().getRhythmConnected() ? OnOffType.ON : OnOffType.OFF);
-        }
+                new HSBType(new DecimalType(hue), new PercentType(saturation), new PercentType(isOn ? brightness : 0)));
+        updateState(CHANNEL_COLOR_MODE, new StringType(state.getColorMode()));
+        updateState(CHANNEL_RHYTHM_ACTIVE,
+                controllerInfo.getRhythm().getRhythmActive() ? OnOffType.ON : OnOffType.OFF);
+        updateState(CHANNEL_RHYTHM_MODE, new DecimalType(controllerInfo.getRhythm().getRhythmMode()));
+        updateState(CHANNEL_RHYTHM_STATE,
+                controllerInfo.getRhythm().getRhythmConnected() ? OnOffType.ON : OnOffType.OFF);
         // update bridge properties which may have changed, or are not present during discovery
         Map<String, String> properties = editProperties();
         properties.put(Thing.PROPERTY_SERIAL_NUMBER, controllerInfo.getSerialNo());
         properties.put(Thing.PROPERTY_FIRMWARE_VERSION, controllerInfo.getFirmwareVersion());
         updateProperties(properties);
 
+        Configuration config = editConfiguration();
+
+        if (controllerInfo.getModel().equals(MODEL_ID_CANVAS)) {
+            config.put(NanoleafControllerConfig.DEVICE_TYPE,DEVICE_TYPE_CANVAS);
+            logger.debug("Set to device type {}", DEVICE_TYPE_CANVAS);
+        } else {
+            config.put(NanoleafControllerConfig.DEVICE_TYPE,DEVICE_TYPE_LIGHTPANELS);
+            logger.debug("Set to device type {}", DEVICE_TYPE_LIGHTPANELS);
+        }
+        updateConfiguration(config);
+
+        getConfig().getProperties().forEach(

```

Comment:
Wrapping these two forEach statements in a `if (logger.isTraceEnabled()) { ... }` please